dialoptions.go

```diff
@@ -557,7 +568,7 @@ func withHealthCheckFunc(f internal.HealthChecker) DialOption {
 	})
 }
 
-func defaultDialOptions() dialOptions {
+func newDefaultDialOptions() dialOptions {
```

Comment:
Why the change here?

Reply:
Reverted.

You mentioned it's a bit confusing around the naming, and update the name helps to differentiate the variable and the factory method.